### PR TITLE
2021 updates

### PR DESCRIPTION
**Pull request by gxavier38**

### CODE REVIEW

File: server/app/sessionStatus.js [hunk 1]
```diff
@@ -98,7 +100,7 @@ module.exports.getSubmissionHistory = function (context, body, res) {
           arrNoNulls[i - shift] = current;
         }
       }
-      to_send[cohort] = {history: arrNoNulls.slice(0, arrNoNulls.length - shift), count: count};
+      to_send[cohort] = {history: arrNoNulls.slice(0, arrNoNulls.length - shift), count: count, data_1: data[1], data_0:data[0] };
```
Comment (KinanBab) [inline server/app/sessionStatus.js]: I am not 100% sure if I am correct, but it seems like the userkey is being sent next to the date of submission back to the analyst. if thats the case this is a problem: the analyst will be able to tell who submitted when. Furthermore, it is important that the analyst can never link the userkey (which they know which of the input companies it belongs to) to the jiff party_id which is randomly selected by the server, and only the server knows to which userkey it belongs to.

This here could be an attack vector, even if the information is not displayed to the analyst, if the analyst can use a browser inspector to view incoming data etc.

I am not sure what data_0 and data_1 are.

It is ok to use the userkey server side to find the most recent submission etc. But it is important that it is removed from the array before it is sent to the analyst. The easiest way to check this is to log the response analyst side and make sure it is anonymous.

File: server/jiff/mailbox.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ module.exports = {
 
       var result = [];
       for (var d of data) {
-        result.push({msg: d.message, label: d.label});
+        result.push({msg: d.message, label: d.label, id: d._id});
```
Comment (KinanBab) [inline server/jiff/mailbox.js:37]: this makes sense

**Comment by KinanBab** [comment]
I left one comment that I think is important security wise.

I am curious, do these changes address the two issues we discussed earlier over our call? the test not detecting the cells beings filled properly, and re-submission. The reason I ask is I could not find concrete changes that i thought corresponded to them

File: server/app/sessionStatus.js (hunk 1 above)
Comment (gxavier38) [reply server/app/sessionStatus.js]: I think this was a feature request the BWWC made, and they accepted the privacy tradeoff for the ability to check who hasn't submitted data. 

Do you problems other than knowing which parties already submitted?

**Comment by gxavier38** [comment]
> do these changes address the two issues we discussed earlier over our call? the test not detecting the cells beings filled properly, and re-submission

Those changes are mainly in JIFF at https://github.com/multiparty/jiff/pull/226.

File: server/app/sessionStatus.js (hunk 1 above)
Comment (vidyaap) [reply server/app/sessionStatus.js]: Can confirm, this was requested by BWWC. We could probably make this configurable while we merge the pacesetters version too.

Comment (KinanBab) [reply server/app/sessionStatus.js]: OK, if it is a feature request it is what it is. I do not think there are other problems. I looked and the associated jiff party id is not revealed or associated with the revealed userkey.

**Comment by KinanBab** [comment]
Makes sense. I will take a look over the other JIFF PR. This PR looks good to me.

**Pull request merged**